rbd/mirror.go

```diff
@@ -71,6 +71,28 @@ func (imm ImageMirrorMode) String() string {
 	}
 }
 
+//  GetMirrorUuid returns the mirroring uuid for the pool associated with the ioctx.
+//
+//  Implements:
+//  int rbd_mirror_uuid_get(rados_ioctx_t io_ctx,
+//	                       char *uuid, size_t *max_len);
+func GetMirrorUuid(ioctx *rados.IOContext) (string, error) {
+	var (
+		err   error
+		uuid  []byte
+		cSize C.size_t
+	)
+	ret := C.rbd_mirror_uuid_get(
+		cephIoctx(ioctx),
+		(*C.char)(unsafe.Pointer(&uuid[0])),
```

Comment:
This line can't currently work as intended, because the uuid slice isn't backed by any array/data. Above, uuid is declared as a []byte (byte-slice) but this is like a nil. The slice we're "casting "into a c string C needs to have enough memory to hold the data the function wants to write into it (I often call this a buffer as a shorthand). The C implementation returns an ERANGE errno https://github.com/ceph/ceph/blob/77edcfdcdd732ce75987035a46d8c9a560e57b34/src/librbd/librbd.cc#L3174  if it can't fit.
We have a retry mechanism in the code base to handle such situations. Example:
https://github.com/ceph/go-ceph/blob/78f2f30f63419032cf41b6c669935fc089784109/rbd/mirror.go#L181

Generally, UUIDs are all the same size, so we can try to make the initial size of the retry call big enough so it only needs to try once, but I would still use the retry function just in case. :-)

Reply:
OK, thanks, I read about this retry mechanism in other places in the code base but couldn't really understand how can I use it here, but now I can. Will update.

Reply:
Updated!. PTAL, thanks!.